packages/merge-styles/src/IStyleSet.ts

```diff
@@ -25,7 +25,7 @@ export type __MapToFunctionType<T> = Extract<T, Function> extends never
 export type IStyleSet<TStyleSet extends IStyleSet<TStyleSet>> = {
   [P in keyof Omit<TStyleSet, 'subComponentStyles'>]: IStyle;
 } & {
-  subComponentStyles?: { [P in keyof TStyleSet['subComponentStyles']]: IStyleFunctionOrObject<any, IStyleSet<any>> };
+  subComponentStyles?: { [P in keyof TStyleSet['subComponentStyles']]: IStyleFunctionOrObject<any, any> };
```

Comment:
Why was this wrong?

Reply:
After adding `subComponentStyles: IPanelSubComponentStyles;` in `IPanelStyle`, i got this error on `IDropdownStyles`:
```
Type 'IDropdownStyles' does not satisfy the constraint 'IStyleSet<IDropdownStyles>'.
  Type 'IDropdownStyles' is not assignable to type '{ subComponentStyles?: { panel: IStyleFunctionOrObject<any, IStyleSet<any>>; label: IStyleFunctionOrObject<any, IStyleSet<any>>; multiSelectItem: IStyleFunctionOrObject<...>; } | undefined; }'.
    The types of 'subComponentStyles.panel' are incompatible between these types.
      Type 'IStyleFunctionOrObject<IPanelStyleProps, IPanelStyles>' is not assignable to type 'IStyleFunctionOrObject<any, IStyleSet<any>>'.
        Type 'IStyleFunction<IPanelStyleProps, IPanelStyles>' is not assignable to type 'IStyleFunctionOrObject<any, IStyleSet<any>>'.
          Type 'IStyleFunction<IPanelStyleProps, IPanelStyles>' is not assignable to type 'IStyleFunction<any, IStyleSet<any>>'.
            Call signature return types 'Partial<IPanelStyles>' and 'Partial<IStyleSet<any>>' are incompatible.
              The types of 'subComponentStyles' are incompatible between these types.
                Type 'IPanelSubComponentStyles | undefined' is not assignable to type '{ [x: string]: IStyleFunctionOrObject<any, IStyleSet<any>>; } | undefined'.
                  Type 'IPanelSubComponentStyles' is not assignable to type '{ [x: string]: IStyleFunctionOrObject<any, IStyleSet<any>>; }'.
                    Index signature is missing in type 'IPanelSubComponentStyles'.
```

Honestly, I don't understand exactly why this error. would be great if you know here. 
but note having `IStyleFunctionOrObject<any, any>` is still type safe bc:
```
export declare type IStyleFunctionOrObject<TStylesProps, TStyleSet extends IStyleSet<TStyleSet>> 
```
here generic type `TStyleSet` extends `IStyleSet`

Reply:
It's really weird that the use of `IStyleFunctionOrObject` would impose an interface requirement on a type arg. That should be left up to `IStyleFunctionOrObject`'s definition, which it seems to already do in this case. Like Xu said, `IStyleSet<any>` appears redundant since `IStyleFunctionOrObject` is requiring it on the second type argument.

Reply:
In that case I'm okay with it too